### PR TITLE
Add GitLab pipeline

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

**Review by jkwatson — APPROVED** [review]
I will be doing the same for Android soon

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  name: "openjdk:11.0.11-9-jdk"
+
+stages:
+  - build
+  - release
+
+build:
+  stage: build
+  script:
+    - ./mvnw clean install -DskipTests
```
Comment (laurit) [inline .gitlab-ci.yml:11]: github uses `./mvnw clean package -DskipTests` and `./mvnw verify` any reason why it is different here? Before https://github.com/signalfx/signalfx-java/pull/119 github used the same commands as here.

**Review by laurit — APPROVED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (mateuszrzeszutek) [reply .gitlab-ci.yml:11]: No reason -- this one can use `package` too; I just typed those commands from the memory and not copied them 😅

Comment (laurit) [reply .gitlab-ci.yml]: Actually github uses `./mvnw test` instead of `./mvnw verify`, meant to write it in the first comment but messed up :)

Comment (mateuszrzeszutek) [reply .gitlab-ci.yml]: Okay, now they're the same 😛

Comment (jkwatson) [reply .gitlab-ci.yml]: I *think* `verify` is the preferred maven way to do things now, FYI.

Comment (mateuszrzeszutek) [reply .gitlab-ci.yml]: OK, let's use verify everywhere then 👍

**Pull request merged**